src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/MemoryCacheEntry.cs

```diff
@@ -240,7 +240,11 @@ private void OnDependencyChanged(object state)
         {
             if (State == EntryState.AddedToCache)
             {
-                _fields._cache.RemoveEntry(this.Key, this, CacheEntryRemovedReason.ChangeMonitorChanged);
+                try
+                {
+                    _fields._cache.RemoveEntry(this.Key, this, CacheEntryRemovedReason.ChangeMonitorChanged);
+                }
+                catch (ObjectDisposedException) { }
             }
```

Comment:
I couldn't work out where ObjectDisposedException would even be thrown. Throwing and catching an exception is expensive. Is there a way to suppress this? Add an IsDisposed check to do a best effort attempt to avoid it being thrown for example?